src/build.d

```diff
@@ -525,10 +525,12 @@ alias buildFrontendHeaders = makeRule!((builder, rule) {
     builder
         .name("cxx-headers")
         .description("Build the C++ frontend headers ")
-        .msg("(DC) CXX-HEADERS")
+        .msg("(DMD) CXX-HEADERS")
         .deps([dmdDefault])
         .target(env["G"].buildPath("frontend.h"))
-        .command([dmdExeFile] ~ flags["DFLAGS"] ~
+        .command([dmdExeFile] ~
+            flags["DFLAGS"]
+              .filter!(f => matchFirst(f, ctRegex!`^-(debug=|I|J)`)).array ~
```

Comment:
This also removes `-version=...`.
Remove sanitizer flags instead.

Reply:
Removing sanitizer flags doesn't solve the problem completely. Sanitizer options is only one of the LDC specific flags. What if there is other LDC/GDC flags, such as specific optimization/transformation passes? I think we should have an allowlist instead, since this is restricted to header generation, otherwise denying every specific flag is impractical.

Reply:
Fair enough.